### ISSUE TITLE
Odd global certificates

### ISSUE THREAD

**Issue by 0xDiBa** [mozilla/TLS-Observer#13]
There seem to be root non CA certificates validating domains directly.
Possibly they are used for backwards compatibility.

![odd_cert](https://cloud.githubusercontent.com/assets/5148180/5725453/99471780-9b5b-11e4-827f-5979381b9a2e.png)

**Comment by jvehent** [comment]
Most likely a bug in the chain building. The cert above is a valid verisign root.

```
$ openssl s_client -connect 72.21.215.232:443 -showcerts -CAfile /etc/ssl/certs/ca-bundle.crt 
CONNECTED(00000003)
depth=3 C = US, O = "VeriSign, Inc.", OU = Class 3 Public Primary Certification Authority
verify return:1
depth=2 C = US, O = "VeriSign, Inc.", OU = VeriSign Trust Network, OU = "(c) 2006 VeriSign, Inc. - For authorized use only", CN = VeriSign Class 3 Public Primary Certification Authority - G5
verify return:1
depth=1 C = US, O = "VeriSign, Inc.", OU = VeriSign Trust Network, OU = Terms of use at https://www.verisign.com/rpa (c)10, CN = VeriSign Class 3 Secure Server CA - G3
verify return:1
depth=0 C = US, ST = Washington, L = Seattle, O = Amazon.com Inc., CN = www.amazon.com
verify return:1
```

**Comment by 0xDiBa** [comment]
There is some kind of mixup in the Ubuntu truststore ( both 12.04 & 14.04 ). 
Look at the chains I am getting when validating against amazon.com:

    2015/01/18 13:18:55 --------- Received --------
    2015/01/18 13:18:55 www.amazon.com
    2015/01/18 13:18:55 VeriSign Class 3 Secure Server CA - G3
    2015/01/18 13:18:55 VeriSign Class 3 Public Primary Certification Authority - G5
    2015/01/18 13:18:55 -> Truststore : NSS
    2015/01/18 13:18:55 Trust chain no:0
    2015/01/18 13:18:55 Subject :www.amazon.com
    2015/01/18 13:18:55 Issuer :VeriSign Class 3 Secure Server CA - G3
    2015/01/18 13:18:56 record not found
    2015/01/18 13:18:56 ISCA :false
    2015/01/18 13:18:56 Appending domain :amazon.com
    2015/01/18 13:18:57 Subject :VeriSign Class 3 Secure Server CA - G3
    2015/01/18 13:18:57 Issuer :VeriSign Class 3 Public Primary Certification Authority - G5
    2015/01/18 13:18:57 ISCA :true
    2015/01/18 13:18:57 Subject :VeriSign Class 3 Public Primary Certification Authority - G5
    2015/01/18 13:18:57 Issuer :VeriSign Class 3 Public Primary Certification Authority - G5
    2015/01/18 13:18:57 ISCA :true
    2015/01/18 13:18:57 -> Truststore : Ubuntu-14.04
    2015/01/18 13:18:57 Trust chain no:0
    2015/01/18 13:18:57 Subject :www.amazon.com
    2015/01/18 13:18:57 Issuer :VeriSign Class 3 Secure Server CA - G3
    2015/01/18 13:18:58 Subject :VeriSign Class 3 Secure Server CA - G3
    2015/01/18 13:18:58 Issuer :VeriSign Class 3 Public Primary Certification Authority - G5
    2015/01/18 13:18:58 Subject :VeriSign Class 3 Public Primary Certification Authority - G5
    2015/01/18 13:18:58 Issuer :VeriSign Class 3 Public Primary Certification Authority - G5
    2015/01/18 13:18:58 Trust chain no:1
    2015/01/18 13:18:58 Subject :www.amazon.com
    2015/01/18 13:18:58 Issuer :VeriSign Class 3 Secure Server CA - G3
    2015/01/18 13:18:59 Subject :VeriSign Class 3 Secure Server CA - G3
    2015/01/18 13:18:59 Issuer :VeriSign Class 3 Public Primary Certification Authority - G5
    2015/01/18 13:18:59 Subject :VeriSign Class 3 Public Primary Certification Authority - G5
    2015/01/18 13:18:59 Issuer :
    2015/01/18 13:18:59 ISCA :true
    2015/01/18 13:18:59 Subject :
    2015/01/18 13:18:59 Issuer :
    2015/01/18 13:18:59 ISCA :false
    2015/01/18 13:18:59 Appending domain :amazon.com
    2015/01/18 13:19:00 Trust chain no:2
    2015/01/18 13:19:00 Subject :www.amazon.com
    2015/01/18 13:19:00 Issuer :VeriSign Class 3 Secure Server CA - G3
    2015/01/18 13:19:00 Subject :VeriSign Class 3 Secure Server CA - G3
    2015/01/18 13:19:00 Issuer :VeriSign Class 3 Public Primary Certification Authority - G5
    2015/01/18 13:19:00 Subject :VeriSign Class 3 Public Primary Certification Authority - G5
    2015/01/18 13:19:00 Issuer :
    2015/01/18 13:19:00 Subject :
    2015/01/18 13:19:00 Issuer :
    2015/01/18 13:19:01 ISCA :false
    2015/01/18 13:19:01 Appending domain :amazon.com

**Comment by jvehent** [comment]
Those certificates are most likely X509 V1 or V2, which doesn't indicate the x509v3 extension that a certificate is a CA. We should check certificates that are V1 or V2 against the trustores, and if a certificate exists in a trustore, manually set the CA=true bit.

**Comment by 0xDiBa** [comment]
Fixed with f89bece3fe07acbcc9a6c87f17be822f5fb31292
